docs/boot_modes.md

```diff
@@ -0,0 +1,111 @@
+# GardenLinux Boot Modes :potted_plant::penguin:
+
+GardenLinux supports two primary boot modes: Legacy and Unified System Image (USI).
+
+Legacy is the traditional way most Linux distros boot and also how all GardenLinux versions prior to 17?? used to boot.
+Legacy boot images are dual mode images, working both with UEFI boot and BIOS boot.
+
+USI mode is a novel approach, where the entire system gets embedded into a single UKI binary.
+This brings several advantages, such as allowing for in-place updates and enhanced security, but is only compatible with UEFI boot.
+
+|                          | USI                | Legacy             |
+|--------------------------|--------------------|--------------------|
+| UEFI boot                | :white_check_mark: | :white_check_mark: |
+| BIOS boot                | :x:                | :white_check_mark: |
+| secureboot / trustedboot | :heavy_check_mark: | :x:                |
+| in-place updates         | :white_check_mark: | :x:                |
+| TPM 2.0 disk encryption  | :heavy_check_mark: | :x:                |
+
+*( :heavy_check_mark: = optional )*
+
+> [!TIP]
+> If you are unsure which mode to pick: go for the *trustedboot* images, these are USIs with all optional security features enabled.
```

Comment:
who does this address? is this specific to gardener? what about users who build custom images based on Garden Linux?

Reply:
not specific to gardener, but to anyone consuming pre-build images. If you build your own images you won't get around reading the whole doc to decide what you want to build.